### DISCUSSION TITLE
Zigbee2Mqtt only runs after install, after reboot no more connection

### DISCUSSION THREAD

**Discussion by GertonWullink** [Koenkk/zigbee2mqtt#18378] [General]
Mij RPi with lateste Openhab, wont run Zigbee2mqtt after reboot. 

at first, after install it runs like a charm, configuration page is accessibel. 
after first reboot of Pi, the zigbee2mqtt is down, also page 8081 cant 
connect. 

Can u help me?

**Comment by lux73** [comment]
what does

`sudo service zigbee2mqtt status`

show?

**Reply by GertonWullink** [reply]
● zigbee2mqtt.service - Zigbee2MQTT
     Loaded: loaded (/etc/systemd/system/zigbee2mqtt.service; enabled; vendor preset: enabled)
     Active: active (running) since Fri 2023-07-21 06:59:47 CEST; 10s ago
   Main PID: 19612 (npm start)
      Tasks: 19 (limit: 2176)
        CPU: 12.924s
     CGroup: /system.slice/zigbee2mqtt.service
             ├─19612 npm start
             ├─19624 sh -c node index.js
             └─19625 node index.js

Jul 21 06:59:47 openhabian systemd[1]: Started Zigbee2MQTT.
Jul 21 06:59:51 openhabian npm[19612]: > zigbee2mqtt@1.32.1 start
Jul 21 06:59:51 openhabian npm[19612]: > node index.js


Time since running isnt logical.... rebooted aprox 10 min. ago.

**Reply by lux73** [reply]
all seems good Service is running now

please do a reboot and check again

**Reply by GertonWullink** [reply]
Rebooted, nothing has changed

This is the output when its running direct after install:

openhabian@openhabian:~ $ ● zigbee2mqtt.service - Zigbee2MQTT
     Loaded: loaded (/etc/systemd/system/zigbee2mqtt.service; enabled; vendor preset: enabled)
     Active: active (running) since Fri 2023-07-21 07:16:47 CEST; 2min 21s ago
   Main PID: 21608 (npm start)
      Tasks: 23 (limit: 2176)
        CPU: 3min 7.315s
     CGroup: /system.slice/zigbee2mqtt.service
             ├─21608 npm start
             ├─21620 sh -c node index.js
             └─21621 node index.js

Jul 21 07:16:47 openhabian systemd[1]: Started Zigbee2MQTT.
Jul 21 07:16:51 openhabian npm[21608]: > zigbee2mqtt@1.32.1 start
Jul 21 07:16:51 openhabian npm[21608]: > node index.js
Jul 21 07:18:11 openhabian npm[21621]: Building Zigbee2MQTT... (initial build), finished
Jul 21 07:18:44 openhabian npm[21621]: Zigbee2MQTT:warn  2023-07-21 07:18:44: `permit_join` set to  `true` in configuration.yaml.
Jul 21 07:18:44 openhabian npm[21621]: Zigbee2MQTT:warn  2023-07-21 07:18:44: Allowing new devices to join.
Jul 21 07:18:44 openhabian npm[21621]: Zigbee2MQTT:warn  2023-07-21 07:18:44: Set `permit_join` to `false` once you joined all devices.

Now i have tried several times:

openhabian@openhabian:~ $ sudo service zigbee2mqtt status
● zigbee2mqtt.service - Zigbee2MQTT
     Loaded: loaded (/etc/systemd/system/zigbee2mqtt.service; enabled; vendor preset: enabled)
     Active: active (running) since Fri 2023-07-21 08:52:32 CEST; 3s ago
   Main PID: 2160 (npm start)
      Tasks: 13 (limit: 2176)
        CPU: 3.305s
     CGroup: /system.slice/zigbee2mqtt.service
             ├─2160 npm start
             ├─2185 sh -c node index.js
             └─2186 node index.js

Jul 21 08:52:32 openhabian systemd[1]: Started Zigbee2MQTT.
Jul 21 08:52:35 openhabian npm[2160]: > zigbee2mqtt@1.32.1 start
Jul 21 08:52:35 openhabian npm[2160]: > node index.js
openhabian@openhabian:~ $ sudo service zigbee2mqtt status
● zigbee2mqtt.service - Zigbee2MQTT
     Loaded: loaded (/etc/systemd/system/zigbee2mqtt.service; enabled; vendor preset: enabled)
     Active: active (running) since Fri 2023-07-21 08:52:47 CEST; 1s ago
   Main PID: 2245 (npm)
      Tasks: 11 (limit: 2176)
        CPU: 2.197s
     CGroup: /system.slice/zigbee2mqtt.service
             └─2245 npm

Jul 21 08:52:47 openhabian systemd[1]: Started Zigbee2MQTT.
openhabian@openhabian:~ $ sudo service zigbee2mqtt status
● zigbee2mqtt.service - Zigbee2MQTT
     Loaded: loaded (/etc/systemd/system/zigbee2mqtt.service; enabled; vendor preset: enabled)
     Active: active (running) since Fri 2023-07-21 08:52:53 CEST; 3s ago
   Main PID: 2273 (npm start)
      Tasks: 19 (limit: 2176)
        CPU: 4.433s
     CGroup: /system.slice/zigbee2mqtt.service
             ├─2273 npm start
             ├─2284 sh -c node index.js
             └─2285 node index.js

Jul 21 08:52:53 openhabian systemd[1]: Started Zigbee2MQTT.
Jul 21 08:52:55 openhabian npm[2273]: > zigbee2mqtt@1.32.1 start
Jul 21 08:52:55 openhabian npm[2273]: > node index.js
openhabian@openhabian:~ $ sudo service zigbee2mqtt status
● zigbee2mqtt.service - Zigbee2MQTT
     Loaded: loaded (/etc/systemd/system/zigbee2mqtt.service; enabled; vendor preset: enabled)
     Active: active (running) since Fri 2023-07-21 08:53:05 CEST; 505ms ago
   Main PID: 2327 (node)
      Tasks: 7 (limit: 2176)
        CPU: 491ms
     CGroup: /system.slice/zigbee2mqtt.service
             └─2327 node /usr/bin/npm start

Jul 21 08:53:05 openhabian systemd[1]: Started Zigbee2MQTT.

Every time its telling me its running since af few seconds, Thats wy i think its rebooting continuesly

**Reply by lux73** [reply]
Ah okay, understood

if you start manual via `npm start` inside your z2m Folder it works?

**Reply by GertonWullink** [reply]
manual start give me:

/opt/zigbee2mqtt$ npm start

> zigbee2mqtt@1.32.1 start
> node index.js


/opt/zigbee2mqtt/node_modules/js-yaml/lib/loader.js:183
  return new YAMLException(message, mark);
         ^
YAMLException: duplicated mapping key (85:1)

 82 |   '0x00158d0006c5112a':
 83 |     friendly_name: Aqara Motion ...
 84 |     description: Bovenste plank ...
 85 | advanced:
------^
 86 |   log_level: debug
    at generateError (/opt/zigbee2mqtt/node_modules/js-yaml/lib/loader.js:183:10)
    at throwError (/opt/zigbee2mqtt/node_modules/js-yaml/lib/loader.js:187:9)
    at storeMappingPair (/opt/zigbee2mqtt/node_modules/js-yaml/lib/loader.js:358:7)
    at readBlockMapping (/opt/zigbee2mqtt/node_modules/js-yaml/lib/loader.js:1173:9)
    at composeNode (/opt/zigbee2mqtt/node_modules/js-yaml/lib/loader.js:1441:12)
    at readDocument (/opt/zigbee2mqtt/node_modules/js-yaml/lib/loader.js:1625:3)
    at loadDocuments (/opt/zigbee2mqtt/node_modules/js-yaml/lib/loader.js:1688:5)
    at Object.load (/opt/zigbee2mqtt/node_modules/js-yaml/lib/loader.js:1714:19)
    at Object.read (/opt/zigbee2mqtt/lib/util/yaml.ts:7:21)
    at read (/opt/zigbee2mqtt/lib/util/settings.ts:328:20)

**Reply by lux73** [reply]
I think there is a Problem with you configuration.yaml

either not readable - incorrect user permission or false formatting inside

Please  note .yaml is very sensitive for incorrect spaces...

show your config pls

**Reply by GertonWullink** [reply]
sorry, this log is from after revert a backup i made.

**Reply by GertonWullink** [reply]
/opt/zigbee2mqtt$ npm start

> zigbee2mqtt@1.32.1 start
> node index.js

Zigbee2MQTT:error 2023-07-21 09:31:57: Error while starting zigbee-herdsman
Zigbee2MQTT:error 2023-07-21 09:31:57: Failed to start zigbee
Zigbee2MQTT:error 2023-07-21 09:31:58: Check https://www.zigbee2mqtt.io/guide/installation/20_zigbee2mqtt-fails-to-start.html for possible solutions
Zigbee2MQTT:error 2023-07-21 09:31:58: Exiting...
Zigbee2MQTT:error 2023-07-21 09:31:58: Error: Failed to connect to the adapter (Error: SRSP - SYS - ping after 6000ms)
    at ZStackAdapter.start (/opt/zigbee2mqtt/node_modules/zigbee-herdsman/src/adapter/z-stack/adapter/zStackAdapter.ts:103:27)
    at Controller.start (/opt/zigbee2mqtt/node_modules/zigbee-herdsman/src/controller/controller.ts:132:29)
    at Zigbee.start (/opt/zigbee2mqtt/lib/zigbee.ts:58:27)
    at Controller.start (/opt/zigbee2mqtt/lib/controller.ts:101:27)
    at start (/opt/zigbee2mqtt/index.js:107:5)

After reinstall and flashing Coordinator with latest [CC2652RB_coordinator_20230507.zip] (https://github.com/Koenkk/Z-Stack-firmware/blob/master/coordinator/Z-Stack_3.x.0/bin/CC2652RB_coordinator_20230507.zip)

it looks like its only getting worse...

**Reply by lux73** [reply]
Connection to your Coordinator failed this time

`sudo lsusb`

**Reply by GertonWullink** [reply]
I think i am getting there, 

First i flashed the right firmware for my launchpad dev board. 
i made a mistake. https://github.com/Koenkk/Z-Stack-firmware/blob/master/coordinator/Z-Stack_3.x.0/bin/CC2652R_coordinator_20230507.zip 

flashing this did the trick, now i can reboot and it keeps working. 
i am going to pair my sensors en see if it keeps working. 

Thanks for the help!!!

openhabian@openhabian:~ $ sudo service zigbee2mqtt status
[sudo] password for openhabian:
● zigbee2mqtt.service - Zigbee2MQTT
     Loaded: loaded (/etc/systemd/system/zigbee2mqtt.service; enabled; vendor p>
     Active: active (running) since Fri 2023-07-21 10:12:03 CEST; 58s ago
   Main PID: 920 (npm start)
      Tasks: 23 (limit: 2176)
        CPU: 30.039s
     CGroup: /system.slice/zigbee2mqtt.service
             ├─920 npm start
             ├─936 sh -c node index.js
             └─937 node index.js

Jul 21 10:12:03 openhabian systemd[1]: Started Zigbee2MQTT.
Jul 21 10:12:07 openhabian npm[920]: > zigbee2mqtt@1.32.1 start
Jul 21 10:12:07 openhabian npm[920]: > node index.js
Jul 21 10:12:25 openhabian npm[937]: Zigbee2MQTT:warn  2023-07-21 10:12:25: `pe>
Jul 21 10:12:25 openhabian npm[937]: Zigbee2MQTT:warn  2023-07-21 10:12:25: All>
Jul 21 10:12:25 openhabian npm[937]: Zigbee2MQTT:warn  2023-07-21 10:12:25: Set>
lines 1-17/17 (END)...skipping...
● zigbee2mqtt.service - Zigbee2MQTT
     Loaded: loaded (/etc/systemd/system/zigbee2mqtt.service; enabled; vendor preset: enabled)
     Active: active (running) since Fri 2023-07-21 10:12:03 CEST; 58s ago
   Main PID: 920 (npm start)
      Tasks: 23 (limit: 2176)
        CPU: 30.039s
     CGroup: /system.slice/zigbee2mqtt.service
             ├─920 npm start
             ├─936 sh -c node index.js
             └─937 node index.js

Jul 21 10:12:03 openhabian systemd[1]: Started Zigbee2MQTT.
Jul 21 10:12:07 openhabian npm[920]: > zigbee2mqtt@1.32.1 start
Jul 21 10:12:07 openhabian npm[920]: > node index.js
Jul 21 10:12:25 openhabian npm[937]: Zigbee2MQTT:warn  2023-07-21 10:12:25: `permit_join` set to  `true` in configuration.yaml.
Jul 21 10:12:25 openhabian npm[937]: Zigbee2MQTT:warn  2023-07-21 10:12:25: Allowing new devices to join.
Jul 21 10:12:25 openhabian npm[937]: Zigbee2MQTT:warn  2023-07-21 10:12:25: Set `permit_join` to `false` once you joined all devices.

**Reply by lux73** [reply]
You are lucky that your Coordinator is  working after flashing wrong Firmware

Hava Fun 😀

**Comment by lux73** [comment]
plain Installation or Docker?

**Reply by GertonWullink** [reply]
install on Raspberry pi 4, by configurator from OpenHab (coming from Domoticz and want to try Openhab for the future.
